src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Bjorhus.cpp

```diff
@@ -0,0 +1,469 @@
+    return {};
+  }
+
+  Bjorhus::constraint_preserving_bjorhus_corrections_dt_v_psi(
+      make_not_null(&bc_dt_v_psi), unit_interface_normal_vector,
+      three_index_constraint, char_speeds);
+
+  Bjorhus::constraint_preserving_bjorhus_corrections_dt_v_zero(
+      make_not_null(&bc_dt_v_zero), unit_interface_normal_vector,
+      four_index_constraint, char_speeds);
+
+  if (type_ == detail::ConstraintPreservingBjorhusType::ConstraintPreserving) {
+    Bjorhus::constraint_preserving_bjorhus_corrections_dt_v_minus(
+        make_not_null(&bc_dt_v_minus), gamma2, coords, incoming_null_one_form,
+        outgoing_null_one_form, incoming_null_vector, outgoing_null_vector,
+        projection_ab, projection_Ab, projection_AB,
+        char_projected_rhs_dt_v_psi, char_projected_rhs_dt_v_minus,
+        constraint_char_zero_plus, constraint_char_zero_minus, char_speeds);
+  } else if (type_ == detail::ConstraintPreservingBjorhusType::
+                          ConstraintPreservingPhysical) {
+    Bjorhus::constraint_preserving_physical_bjorhus_corrections_dt_v_minus(
+        make_not_null(&bc_dt_v_minus), gamma2, coords, normal_covector,
+        unit_interface_normal_vector, spacetime_unit_normal_vector,
+        incoming_null_one_form, outgoing_null_one_form, incoming_null_vector,
+        outgoing_null_vector, projection_ab, projection_Ab, projection_AB,
+        inverse_spatial_metric, extrinsic_curvature, spacetime_metric,
+        inverse_spacetime_metric, three_index_constraint,
+        char_projected_rhs_dt_v_psi, char_projected_rhs_dt_v_minus,
+        constraint_char_zero_plus, constraint_char_zero_minus, phi, d_phi, d_pi,
+        char_speeds);
+  } else {
+    ERROR(
+        "Failed to set dtVMinus. Input option must be one of "
+        "ConstraintPreserving or ConstraintPreservingPhysical");
+  }
+
+  // Only add corrections at grid points where the char speeds are negative
+  helpers::set_bc_corr_zero_when_char_speed_is_positive(
+      make_not_null(&bc_dt_v_psi), char_speeds[0]);
+  helpers::set_bc_corr_zero_when_char_speed_is_positive(
+      make_not_null(&bc_dt_v_zero), char_speeds[1]);
+  helpers::set_bc_corr_zero_when_char_speed_is_positive(
+      make_not_null(&bc_dt_v_minus), char_speeds[3]);
+
+  // Convert corrections to dt<evolved variables>
+  auto dt_evolved_vars = evolved_fields_from_characteristic_fields(
+      gamma2, bc_dt_v_psi, bc_dt_v_zero, bc_dt_v_plus, bc_dt_v_minus,
+      normal_covector);
+
+  *dt_pi_correction = get<Tags::Pi<Dim, Frame::Inertial>>(dt_evolved_vars);
+  *dt_phi_correction = get<Tags::Phi<Dim, Frame::Inertial>>(dt_evolved_vars);
+  *dt_spacetime_metric_correction =
+      get<gr::Tags::SpacetimeMetric<Dim, Frame::Inertial, DataVector>>(
+          dt_evolved_vars);
+
+  if (face_mesh_velocity.has_value()) {
+    const auto negative_lambda0 =
+        dot_product(normal_covector, *face_mesh_velocity);
+    if (min(-get(negative_lambda0)) < 0) {
+      return {
```

Comment:
Why doesn't this check the char speed? Why check only negative_lambda0?

Reply:
I added this with the intention of catching the case when the outer boundary is being made to move _outward_, and that is why I am only checking the projection of `face_mesh_velocity` onto the face normal. In case the interface is moving outward we would have to extrapolate data inside of the domain to account for the increasing domain volume which is probably a scenario best avoided. 

Char speeds could legitimately be both positive and negative, so why would we want to return an error message for that?

Reply:
okay, what about renaming the `negative_lambda0` to `radial_mesh_velocity`? Currently it sounds like it's related to the char speeds, which it sort of is, but that's not why you're computing it.

Can `min(-get(...))` be changed to `max(get(...))`? Would that provide the same check? I find the extra negative with the min makes it a bit more difficult to reason about the check. 

 It might also be a good idea to add the `radial_mesh_velocity` to the error message so users can get some idea of what is going on. What do you think?

Reply:
made all these changes, i.e. renamed `negative_lambda0` to `radial_mesh_velocity`, changed `min` to `max`, updated the error message and switched `0` to `0.`.

Reply:
Question for @geoffrey4444 : Since this error was getting triggered for your rotating grid, which should have only tangential motion, should we change the condition here from `max(get(radial_mesh_velocity)) > 0` to `max(get(radial_mesh_velocity)) > 1.e-12` ?

Reply:
I think that makes sense, and I think that depending on what the BBHs do, the tolerance might need to be adjusted down the road